src/firmware/main/math/polynomial_2d_test.cpp

```diff
@@ -0,0 +1,351 @@
+extern "C"
+{
+#include "firmware/main/math/polynomial_2d.h"
+}
+
+#include <gtest/gtest.h>
+
+#include "polynomial_1d.h"
+#include "polynomial_2d.h"
+
+class Polynomial2dTest : public testing::Test
+{
+   protected:
+    virtual void SetUp() {}
+
+    virtual void TearDown() {}
+
+    static void expectRawArraysEq(float* expected, float* actual, size_t num_elements,
+                                  double tolerance = 10e-10)
+    {
+        for (size_t i = 0; i < num_elements; i++)
+        {
+            EXPECT_NEAR(expected[i], actual[i], tolerance)
+                << "expected[" << i << "] is " << expected[i] << " but actual[" << i
+                << "] is " << actual[i];
+        }
+    }
+
+    static void expectArcLengthParametrizationsEqual(ArcLengthParametrization_t expected,
+                                                     ArcLengthParametrization_t actual,
+                                                     double tolerance = 10e-10)
+    {
+        // We can't put asserts in functions, so we have to settle for manually checking
```

Comment:
You mean you can't put asserts in these other Gtest functions? Are you sure? I haven't seen anything that says you can't. The only constraint is that fatal asserts like `ASSERT` can only be used in void-returning functions

Reply:
Ah, forgot they worked in void functions. Changed to an `ASSERT_EQ`